File: data/src/main/java/dev/olog/data/api/lastfm/album/AlbumInfo.java

package dev.olog.data.api.lastfm.album;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class AlbumInfo {

    public Album album;

    public static class Album {

        public String name;

        public String artist;

        public String mbid;

        public String url;

        public List<Image> image = new ArrayList<>();

        public Wiki wiki;

        public static class Image {

            @SerializedName("#text")
            public String text;
            public String size;

        }

        public static class Wiki {

            public String published;
            public String summary;
            public String content;

        }


    }

}

File: data/src/main/java/dev/olog/data/api/lastfm/album/AlbumInfo.kt
package dev.olog.data.api.lastfm.album

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class AlbumInfo(
    val album: Album? = null,
) {

    @Serializable
    class Album(
        val name: String? = null,
        val artist: String? = null,
        val mbid: String? = null,
        val url: String? = null,
        val image: List<Image>? = null,
        val wiki: Wiki? = null,
    )

    @Serializable
    class Image(
        @SerialName("#text")
        val text: String? = null,
        val size: String? = null,
    )

    @Serializable
    class Wiki(
        val published: String? = null,
        val summary: String? = null,
        val content: String? = null,
    )
}